src/poliastro/core/events.py

```diff
@@ -54,3 +54,33 @@ def eclipse_function(k, u_, r_sec, R_sec, R_primary, umbra=True):
     )
 
     return shadow_function
+
+
+@jit
+def line_of_sight(r1, r2, R):
+    """Calculates the line of sight condition between two position vectors, r1 and r2.
+
+    Parameters
+    ----------
+    r1: ~np.array
+        The position vector of the first object with respect to a central attractor.
+    r2: ~np.array
+        The position vector of the second object with respect to a central attractor.
+    R: float
+        The radius of the central attractor.
+
+    Returns
+    -------
+    delta_theta: float
+        Greater than or equal to zero, if there exists a LOS between two objects
+        located by r1 and r2, else negative.
+
+    """
+    r1_norm = np.linalg.norm(r1)
```

Comment:
The condition might be checked here 👍🏽

Reply:
I noticed a slight problem in the edge case we had discussed in #1307. To resolve the problems with any edge cases, the best solution I could think till now is to raise a warning stating "Orbit is going beneath the surface of attractor, LOS event may not work properly".

But we decided a warning was not a good option, right? It might atleast solve the problem.

Reply:
Discussed at lenght in other comments here and in #1307, we decided for the current solution of a warning to minimize complexity